routes/user.go

```diff
@@ -1479,6 +1479,119 @@ func (fes *APIServer) GetHodlersForPublicKey(ww http.ResponseWriter, req *http.R
+	decoder := json.NewDecoder(io.LimitReader(req.Body, MaxRequestBodySizeBytes))
+	requestData := GetBalancesForPublicKeyRequest{}
+	if err := decoder.Decode(&requestData); err != nil {
+		_AddBadRequestError(ww, fmt.Sprintf(
+			"GetHodlersForPublicKey: Problem parsing request body: %v", err))
+		return
+	}
+
+	txnStatus := requestData.TxnStatus
+	if txnStatus == "" {
+		txnStatus = TxnStatusInMempool
+	}
+	// Get a view based on the txnStatus
+	utxoView, err := fes.GetUtxoViewGivenTxnStatus(txnStatus)
+	if err != nil {
+		_AddInternalServerError(ww, fmt.Sprintf("GetDAOCoinLimitOrdersById: Problem fetching utxoView: %v", err))
+		return
+	}
+
+	// Decode the public key for which we are fetching balances
+	if requestData.UserPublicKey == "" {
+		_AddBadRequestError(ww, fmt.Sprintf("GetBalancesForPublicKey: Missing UserPublicKey"))
+		return
+	}
+	userPublicKeyBytes, _, err := lib.Base58CheckDecode(requestData.UserPublicKey)
+	if err != nil {
+		_AddBadRequestError(ww, fmt.Sprintf("GetHodlersForPublicKey: Problem decoding user public key: %v", err))
+		return
+	}
+	if len(requestData.CreatorPublicKeys) == 0 {
+		_AddBadRequestError(ww, fmt.Sprintf("GetBalancesForPublicKey: Missing CreatorPublicKeys"))
+		return
+	}
+
+	balancesMap := make(map[string]*SimpleBalanceResponse)
+	for _, creatorPublicKeyStr := range requestData.CreatorPublicKeys {
+		// Deso is a special case
+		if IsDesoPkid(creatorPublicKeyStr) {
+			desoNanos, err := utxoView.GetDeSoBalanceNanosForPublicKey(userPublicKeyBytes)
+			if err != nil {
+				_AddBadRequestError(ww, fmt.Sprintf("GetBalancesForPublicKey: Problem getting balance: %v", err))
+				return
+			}
+			// When we're dealing with DESO, we use whatever identifier they passed
+			// in as the key. This is the most convenient thing to do for the caller.
+			// If we instead always returned DESO as the key then they would have to
+			// accommodate that, which would be annoying.
+			balancesMap[creatorPublicKeyStr] = &SimpleBalanceResponse{
+				UserPublicKeyBase58Check:    requestData.UserPublicKey,
+				CreatorPublicKeyBase58Check: creatorPublicKeyStr,
+				BalanceBaseUnits:            strconv.FormatUint(desoNanos, 10),
+			}
+			continue
+		}
+		creatorPkBytes, _, err := lib.Base58CheckDecode(creatorPublicKeyStr)
+		if err != nil {
+			_AddBadRequestError(ww, fmt.Sprintf("GetHodlersForPublicKey: Problem decoding creator public key: %v", err))
+			return
+		}
+
+		balanceEntry, _, _ := utxoView.GetBalanceEntryForHODLerPubKeyAndCreatorPubKey(
+			userPublicKeyBytes, creatorPkBytes, true)
+		if balanceEntry == nil || balanceEntry.IsDeleted() {
+			balanceEntry = &lib.BalanceEntry{}
+		}
+		// Convert balanceEntry uint256 to string
+		balancesMap[creatorPublicKeyStr] = &SimpleBalanceResponse{
+			UserPublicKeyBase58Check:    requestData.UserPublicKey,
+			CreatorPublicKeyBase58Check: creatorPublicKeyStr,
+			BalanceBaseUnits:            balanceEntry.BalanceNanos.String(),
```

Comment:
Would returning the hex string here also be useful? So we could do `BalanceBaseUnits` and `BalanceBaseUnitsHex` or something. Not sure if you're concerned about losing precision when going to base10 string.

Reply:
We don't lose precision when we convert to a base-10 string. It's the same value, just easier to read.